runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1135,11 +1149,7 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 
 		if (0 == size) {
 			/* Discontiguous */
-			if (compressObjectReferences()) {
-				size = ((J9IndexableObjectDiscontiguousCompressed *)forwardedHeader->getObject())->size;
-			} else {
-				size = ((J9IndexableObjectDiscontiguousFull *)forwardedHeader->getObject())->size;
-			}
+			size = (uint32_t)getDiscontiguousArraySize((J9IndexableObject *)forwardedHeader->getObject());
```

Comment:
should we not make getContiguousArraySize/getDiscontiguousArraySize return 32bit?

Reply:
Yes, I agree, I also don't like this cast. Please let me see what I can do.